Modernize architecture figures (#1245)
inspektor-gadget/inspektor-gadget: docs/architecture.md

![different tools offered by Inspektor Gadget](images/architecture/ebpf.svg)

> Before using Inspektor Gadget, it first need to be deployed to each node as a
> privileged `DaemonSet`.

Reviewer: I think we should not tell this as a "story" given that this document is not a guide. 

```suggestion
On Kubernetes, Inspektor Gadget is deployed to each node as a DaemonSet.
```

It can be extended with a bit more of information of how it's deployed:

"It's usually deployed by the kubectl gadget plugin. It interacts with the kube-api server to create all the resources needed for Inspektor Gadget to work. Another way of deploying Inspektor Gadget is to directly apply the [manifests](pkg/resources/manifests/deploy.yaml) using kubectl".
